src/quacc/recipes/espresso/bands.py

```diff
@@ -0,0 +1,284 @@
+        to generate some files/info if needed.
+
+    **calc_kwargs
+        Additional keyword arguments to pass to the Espresso calculator. Set a value to
+        `quacc.Remove` to remove a pre-existing key entirely. See the docstring of
+        [quacc.calculators.espresso.espresso.Espresso][] for more information.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
+
+    calc_defaults = {
+        "input_data": {"control": {"calculation": "bands", "verbosity": "high"}}
+    }
+    if primitive:
+        structure = AseAtomsAdaptor.get_structure(atoms)
+        primitive = SpacegroupAnalyzer(structure).get_primitive_standard_structure()
+        atoms = AseAtomsAdaptor.get_atoms(primitive)
+        calc_defaults["kpts"] = bandpath(
+            convert_pmg_kpts({"line_density": 20}, atoms)[0], cell=atoms.get_cell()
+        )
+
+    return base_fn(
+        atoms,
+        template=EspressoTemplate("pw", test_run=test_run),
+        calc_defaults=calc_defaults,
+        calc_swaps=calc_kwargs,
+        parallel_info=parallel_info,
+        additional_fields={"name": "pw.x bands"},
+        copy_files=prev_dir,
+    )
+
+
+def _bands_pp(
+    atoms: Atoms,
+    prev_dir: str | Path,
+    parallel_info: dict[str] | None = None,
+    test_run: bool = False,
+    **calc_kwargs,
+) -> RunSchema:
+    """
+    Function to re-order bands and computes bands-related properties with bands.x.
+
+    Parameters
+    ----------
+    atoms
+        The Atoms object.
+    prev_dir
+        Outdir of the previously ran pw.x calculation. This is used to copy
+        the entire tree structure of that directory to the working directory
+        of this calculation.
+    parallel_info
+        Dictionary containing information about the parallelization of the
+        calculation. See the ASE documentation for more information.
+    test_run
+        If True, a test run is performed to check that the calculation input_data is correct or
+        to generate some files/info if needed.
+
+    **calc_kwargs
+        Additional keyword arguments to pass to the Espresso calculator. Set a value to
+        `quacc.Remove` to remove a pre-existing key entirely. See the docstring of
+        [quacc.calculators.espresso.espresso.Espresso][] for more information.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][].
+        See the type-hint for the data structure.
+    """
+
+    calc_defaults = {}
+
+    return base_fn(
+        atoms,
+        template=EspressoTemplate("bands", test_run=test_run),
+        calc_defaults=calc_defaults,
+        calc_swaps=calc_kwargs,
+        parallel_info=parallel_info,
+        additional_fields={"name": "bands.x post-processing"},
+        copy_files=prev_dir,
+    )
+
+
+def _fermi_surface(
```

Comment:
For this and the other tasks, perhaps what would be best would be to make them all `@job`s (and remove the `_` in front). Then, when they are called within `bands_job`, you can do

```python
from quacc.wflow_tools.customizers import strip_decorator

strip_decorator(fermi_surface)
```

Then one _could_ run the task as a standalone job if they wish. If you do this, feel free to add the jobs to the recipe list too.